File: app/java/app/src/main/java/com/example/accounting/application/exception/ConcurrentModificationException.java
package com.example.accounting.application.exception;

/**
 * 並行修正例外
 *
 * イベントソーシングで楽観的ロックが失敗した場合にスロー
 */
public class ConcurrentModificationException extends RuntimeException {
    public ConcurrentModificationException(String message) {
        super(message);
    }

    public ConcurrentModificationException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: app/csharp/AccountingSystem.Application/Exceptions/ConcurrentModificationException.cs
namespace AccountingSystem.Application.Exceptions;

/// <summary>
/// 同時実行例外
/// イベントソーシングの楽観的ロックで競合が発生した場合にスロー
/// </summary>
public class ConcurrentModificationException : Exception
{
    public ConcurrentModificationException(string message) : base(message)
    {
    }

    public ConcurrentModificationException(string message, Exception innerException)
        : base(message, innerException)
    {
    }
}
